docs: identity provider docs (#5565)
zitadel/zitadel: docs/docs/guides/integrate/identity-providers/azure-ad.md

2. Give the application a name and choose who should be able to login (Single-Tenant, Multi-Tenant, Personal Accounts, etc.) This setting will also have an impact on how to configure the provider later on in ZITADEL.
3. Choose "Web" in the redirect uri field and add the URL:
 - {your-domain}/ui/login/login/externalidp/callback
 - Example redirect url for the domain `https://acme-gzoe4x.zitadel.cloud` would look like this: `https://acme-gzoe4x.zitadel.cloud/ui/login/login/externalidp/callback`
5. Save the Application (client) ID and the Directory (tenant) ID from the detail page

![Azure App Registration](/img/guides/azure_app_registration.png)

![Azure Client ID and Tenant ID](/img/guides/azure_client_tenant_id.png)

### Add client secret

To be able to authenticate your Azure application you have to generate a new client secret.

1. Click on client credentials on the detail page of the application or use the menu "Certificates & secrets"
2. Click on "+ New client secret" and enter a description and an expiry date, add the secret afterwards

> 3. Copy the value of the secret. You will not be able to see the value again afterwards 

Reviewer: ```suggestion
3. Copy the value of the secret. You will not be able to see the value again after some time 
```

![Azure Client Secret](/img/guides/azure_client_secret.png)

### Token configuration

To allow ZITADEL to get the information from the authenticating user you have to configure what kind of optional claims should be returned in the token.

1. Click on Token configuration in the side menu
2. Click on "+ Add optional claim"
3. Add email, family_name, given_name and preferred_username to the id token

![Azure Token configuration](/img/guides/azure_token_configuration.png)

> ### API persmission

Reviewer: Maybe run a spell checker over all files?
```suggestion
### API permissions
```

To be able to get all the information that ZITADEL needs, you have to configure the correct permissions.

1. Go to "API permissions" in the side menu
2. Make sure the permissions include "Microsoft Graph": email, profile and User.Read

> 3. The "Other permissions granted" should include "Microsoft Graph: openid"

Reviewer: lol, interesting :joy:

![Azure API permissions](/img/guides/azure_api_permissions.png)

## ZITADEL Configuration

### Create new Azure AD Provider

Go to the settings of your ZITADEL instance or the organization where you like to add a new **Azure AD** provider.

> Choose the **Microsoft** provider template. This template has everything you need preconfigured. You only have to add the client ID and secret, you have created in the step before.

Reviewer: Let's add (unrendered) line breaks after sentences for better raw markdown readability

```suggestion
Choose the **Microsoft** provider template.
This template has everything you need preconfigured.
You only have to add the client ID and secret, you have created in the step before.
```

You can configure the following settings if you like, a useful default will be filled if you don't change anything:

> **Scopes**: The scopes define which scopes will be sent to the provider, `openid`, `profile`, and `email` are prefilled. This information will be taken to create/update the user within ZITADEL. Make sure to alswo add `User.Read`

Reviewer: ```suggestion
**Scopes**: The scopes define which scopes will be sent to the provider, `openid`, `profile`, and `email` are prefilled.
This information will be taken to create/update the user within ZITADEL. Make sure to also add `User.Read`
```

> **Email Verified**: Azure AD doesn't send the email verified claim in the token of the user, if you don't enable this setting, the user will be created with the email not verified, which results in a Email Verification Message for the user. If you want to avoid that, make sure to enable "Email verified". In that case the user will be created with a verified email address.

Reviewer: I'd use more present tense, split sentences where possible and avoid "of the":

```suggestion
**Email Verified**: Azure AD doesn't send the email verified claim in the users token, if you don't enable this setting.
The user is then created with an unverified email, which results in an email verification message.
If you want to avoid that, make sure to enable "Email verified".
In that case, the user is created with a verified email address.
```

> **Tenant Type**: The tenant type should be configured according what you have chosen in the settings of your Azure Ad application previously.

Reviewer: ```suggestion
**Tenant Type**: Configure the tenant type according to what you have chosen in the settings of your Azure AD application previously.
```

> - Common: Choose common if you want all microsoft accounts being able to login. In this case you have configured "Accounts in any organizational directory and personal Microsoft accounts" in your Azure App configuration

Reviewer: I'd use imperatives wherever possible.

```suggestion
- Common: Choose common if you want all Microsoft accounts being able to login.
In this case, configure "Accounts in any organizational directory and personal Microsoft accounts" in your Azure AD App.
```

- Organizations: Choose organization if you have Azure AD Tenants and no personal accounts. (You have configured either "Accounts in this organization" or "Accounts in any organizational directory" on your Azure APP)
- Consumers: Choose this if you want to allow public accounts. (In your Azure AD App you have configured "Personal Microsoft accounts only")

> **Tenant ID**: If you have selected either "organizations" or "Customers" in the Tenant Type, you have to enter the Directory (Tenant) ID, copied previously in the Azure App configuration, here.

Reviewer: Let's use exact *UI field names in italic*

```suggestion
**Tenant ID**: If you have selected either the *Organizations* or *Customers* as the *Tenant Type*, you have to enter the *Directory (Tenant) ID*, copied previously in the Azure App configuration, here.
```

**Automatic creation**: If this setting is enabled the user will be created automatically within ZITADEL, if it doesn't exist.

**Automatic update**: If this setting is enabled, the user will be updated within ZITADEL, if some user data are changed withing the provider. E.g if the lastname changes on the GitHub account, the information will be changed on the ZITADEL account on the next login. 

**Account creation allowed**: This setting determines if account creation within ZITADEL is allowed or not.

> **Account linking allowed**: This setting determines if account linking is allowed. (E.g an account within ZITADEL should already be existing and the when login with GitHub an account should be linked)

Reviewer: Can we put this in its own mdx file and reuse it in every IdP guide?

Author: Makes sense, I changed it

:::info
Either account creation or account linking have to be enabled. Otherwise, the provider can't be used.
:::

![Azure Provider](/img/guides/zitadel_azure_provider.png)

### Activate IdP

> Once you created the IdP you need to activate it, to make it usable for your users.

Reviewer: ```suggestion
Once you created the IdP you need to activate it.
```

![Activate Azure AD](/img/guides/zitadel_activate_azure.png)

## Test the setup

> To test the setup use incognito mode and browse to your login page.

Reviewer: ```suggestion
To test the setup, use incognito mode and browse to your login page.
```

> If you succeeded you should see a new button which should redirect you to your Microsoft Login.

Reviewer: I don't like should :-)

```suggestion
If you succeed, you see a new button which redirects you to your Microsoft Login.
```

> Per default the login of your instance will be shown, read the following section on how to trigger it for a specific organization: [Organization Scope](./general#trigger-configuration-on-the-login-for-a-specific-organization)

Reviewer: I learned to like long inline links from Matt :-)

```suggestion
By default, your instances login is shown.
Learn more about [how to trigger it for a specific organization](./general#trigger-configuration-on-the-login-for-a-specific-organization).
```
